parquet-column/src/main/java/org/apache/parquet/schema/Type.java

```diff
@@ -333,6 +333,21 @@ void checkContains(Type subType) {
     }
   }
 
+  /**
+   * Checks compatibility with subType
+   * Allows subtype to have a less strict repetition
+   * For eg: a REQUIRED type is compatible with an OPTIONAL subType
+   *
+   * @param subType
+   */
+  void checkCompatibility(Type subType) {
+    if (!this.name.equals(subType.name)
+      || (this.repetition != subType.repetition && !this.repetition.isMoreRestrictiveThan(subType.repetition))) {
```

Comment:
Similarly, unlike in `checkContains` this checks if the `subType`'s `repetition` is more restrictive or not. If it's less restrictive, it's still compatible. This would allow a required field to be projected using an optional field.